app/src/main/java/ch/epfl/cs311/wanderwave/model/data/ProfileTrackAssociation.kt

```diff
@@ -20,13 +20,23 @@ data class ProfileTrackAssociation(
     return hashMapOf("profile" to profile?.toMap(), "track" to track.toMap())
   }
 
+  fun isLiked(profile: Profile): Boolean {
+    return likers.contains(profile)
+  }
+
   fun likeTrack(profile: Profile): ProfileTrackAssociation {
-    this.profile?.numberOfLikes = this.profile?.numberOfLikes!! + 1
+    if (likers.contains(profile)) return this
+    if (this.profile != null) {
+      this.profile.numberOfLikes += 1
+    }
     return ProfileTrackAssociation(profile, track, likers + profile, likes + 1)
   }
 
   fun unlikeTrack(profile: Profile): ProfileTrackAssociation {
-    this.profile?.numberOfLikes = this.profile?.numberOfLikes!! - 1
+    if (!likers.contains(profile)) return this
+    if (this.profile != null) {
+      this.profile.numberOfLikes -= 1
+    }
     return ProfileTrackAssociation(profile, track, likers - profile, likes - 1)
```

Comment:
Sorry for bothering, maybe I'm missing something but wouldn't this return need to be in the if? Because if the profile already like this song we don't want to add it again to the likers no?

Reply:
Yeah that's why there is `if (!likers.contains(profile)) return this` just above, which returns an unchanged ProfileTrackAssociation if the profile has already liked the song

Reply:
I meant adding null to the list, if profile is null we don't want to add it to the list no?